### DISCUSSION TITLE
could EnumIter be recursive ?

### DISCUSSION THREAD

**Discussion by VirgileHenry** [Peternator7/strum#265] [Ideas]
Would it be useful / meaningful to have the EnumIter to work recursively if enum data have this trait too ?
Something like : 

`
#[derive(EnumIter)]

enum E1 {
    A,
    B,
}

#[derive(EnumIter)]
enum E2 {
    C(E1)
}

for v in E2::iter() {
    println!("{v:?}"); // C(A), C(B), 
}
`

**Comment by ActuallyHappening** [comment]
Having the same problem, recursiveness is too restrictive without some attribute
would love there to be an attribute #[strum(recursive)] though for simpler enums

**Comment by juliancoffee** [comment]
I've created an issue (and a PR for this) #424 

So yeah, I guess you could push a like button there/add your use case
